src/core/chat-api.ts

```diff
@@ -103,6 +103,10 @@ export class ChatApi {
 
     // convert the params into internal format
     const apiParams: ApiGetMessagesQueryParams = { ...params };
+    // encode the serial if it is provided
+    if (params.fromSerial) {
+      apiParams.fromSerial = encodeURIComponent(params.fromSerial);
```

Comment:
These are already encoded by `realtime.request`, no need to do it here.

I think doing it here will actually double encode and break decoding?

Reply:
Yeah I think what we have already is fine - we don't do anything special with it for `getPreviousMessages`, which uses the same mechanism and that works fine.